pkg/runner/runner.go

```diff
@@ -406,9 +406,14 @@ func compileAnchoredRegexString(s string) (*regexp.Regexp, error) {
 // said lease; in such and event the connector will be shutdown anyway, and
 // connection-death will close the connection.
 func getSchemaLease(ctx context.Context, conn *pgx.Conn) error {
-	err := util.GetSharedLease(ctx, conn, schemaLockId)
-	if err != nil {
-		return err
+	var (
+		hasSharedLease = true
+		err            = util.GetSharedLease(ctx, conn, schemaLockId)
+	)
+	if err == util.SharedLeaseFailure {
+		hasSharedLease = false
+	} else if err != nil {
+		return fmt.Errorf("get schema lease: %w", err)
 	}
-	return pgmodel.CheckSchemaVersion(ctx, conn, appVersion)
+	return pgmodel.CheckSchemaVersion(ctx, conn, appVersion, hasSharedLease)
```

Comment:
yeah this is not the path we are worried about. The shared lease doesn't matter much